Packs/AzureKubernetesServices/Integrations/AzureKubernetesServices/AzureKubernetesServices.py

```diff
@@ -0,0 +1,194 @@
+                                        f'/{monitoring_resource_name}'
+            else:
+                cluster = self.cluster_get(resource_name)
+                workspace_resource_id = cluster.get('properties', {}).get('addonProfiles', {}).get('omsagent', {})\
+                    .get('config', {}).get('logAnalyticsWorkspaceResourceID')
+            addon_profiles['omsagent'] = {
+                'enabled': monitoring_agent_enabled,
+                'config': {'logAnalyticsWorkspaceResourceID': workspace_resource_id},
+            }
+        return self.ms_client.http_request(
+            'PUT',
+            url_suffix=f'resourceGroups/{self.resource_group_name}/providers/Microsoft.ContainerService/managedClusters'
+                       f'/{resource_name}',
+            params={
+                'api-version': API_VERSION,
+            },
+            json_data={
+                'location': location,
+                'properties': {
+                    'addonProfiles': addon_profiles,
+                }
+            },
+            timeout=30,
+        )
+
+
+def clusters_list(client: AKSClient) -> CommandResults:
+    response = client.clusters_list_request()
+    clusters = response.get('value', [])
+    readable_output = [{
+        'Name': cluster.get('name'),
+        'Status': cluster.get('properties', {}).get('provisioningState'),
+        'Location': cluster.get('location'),
+        'Tags': cluster.get('tags'),
+        'Kubernetes version': cluster.get('properties', {}).get('kubernetesVersion'),
+        'API server address': cluster.get('properties', {}).get('fqdn'),
+        'Network type (plugin)': cluster.get('properties', {}).get('networkProfile', {}).get('networkPlugin'),
+    } for cluster in clusters]
+    return CommandResults(
+        outputs_prefix='AzureKS.ManagedCluster',
+        outputs_key_field='id',
+        outputs=clusters,
+        readable_output=tableToMarkdown(
+            'AKS Clusters List',
+            readable_output,
+            ['Name', 'Status', 'Location', 'Tags', 'Kubernetes version', 'API server address', 'Network type (plugin)'],
+        ),
+        raw_response=response,
+    )
+
+
+def clusters_addon_update(client: AKSClient, args: Dict) -> str:
+    update_args = {
+        'resource_name': args.get('resource_name'),
+        'location': args.get('location'),
+    }
+    if args.get('http_application_routing_enabled'):
+        update_args['http_application_routing_enabled'] = argToBoolean(args.get('http_application_routing_enabled'))
+    if args.get('monitoring_agent_enabled'):
+        update_args['monitoring_agent_enabled'] = argToBoolean(args.get('monitoring_agent_enabled'))
+        update_args['monitoring_resource_name'] = args.get('monitoring_resource_name')
+    client.cluster_addon_update(**update_args)
+    return 'The request to update the managed cluster was sent successfully.'
+
+
+def start_auth(client: AKSClient) -> CommandResults:
```

Comment:
after finishing the authentication process (start + complete), if one runs the `start` command again, will he also have to run the `complete` or will it keep using the saved access token?

Reply:
user can run `start` again but without running `complete` after it, it wont have any effect and it will keep using the cached token